src/native/mac/qgsmacnative.mm

```diff
@@ -41,3 +74,38 @@
   NSArray *fileURLs = [NSArray arrayWithObjects:[NSURL fileURLWithPath:pathStr], nil];
   [[NSWorkspace sharedWorkspace] activateFileViewerSelectingURLs:fileURLs];
 }
+
+QgsNative::Capabilities QgsMacNative::capabilities() const
+{
+  return NativeDesktopNotifications;
+}
+
+QgsNative::NotificationResult QgsMacNative::showDesktopNotification( const QString &summary,
+    const QString &body,
+    const QgsNative::NotificationSettings &settings )
+{
+  NSUserNotification *notification = [[NSUserNotification alloc] init];
+  notification.title = summary.toNSString();
+  notification.informativeText = body.toNSString();
+  notification.soundName = NSUserNotificationDefaultSoundName;   //Will play a default sound
```

Comment:
Is this normal behaviour on osx? Or will a sound become annoying quickly?

Reply:
yep, but then you can disable them in the notifications settings

![image](https://user-images.githubusercontent.com/127259/44443442-59a45780-a5e0-11e8-839a-b657757449b5.png)